hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/storage/HoodieHFileWriter.java

```diff
@@ -122,7 +128,13 @@ public boolean canWrite() {
 
   @Override
   public void writeAvro(String recordKey, IndexedRecord object) throws IOException {
-    byte[] value = HoodieAvroUtils.avroToBytes((GenericRecord)object);
+    byte[] value = HoodieAvroUtils.avroToBytes((GenericRecord) object);
+    if (schemaRecordKeyField.isPresent()) {
+      GenericRecord recordKeyExcludedRecord = HoodieAvroUtils.bytesToAvro(value, this.schema);
```

Comment:
This will reduce performance as you are converting the record to bytes in the line above and then immediately parsing it back to the GenericRecord again. 

If may be better to check first before creating the bytes.

Reply:
Added the check for non empty key string before doing this. Let me know if you are referring to some other additional check here.